### PR TITLE
Announcement Listener

### PR DESCRIPTION
**Pull request by ShaneAB**
Mirrors announcements from a discord channel and copies them into the specified database.

### CODE REVIEW

File: src/bot/AnnouncementListener.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import * as Discord from "discord.js";
+import * as MySQL from "mysql2/promise";
+import { IBotConfig } from "../api";
+
+export class AnnouncmentListener {
```
Comment (tomrijnbeek) [inline src/bot/AnnouncementListener.ts:5]: Typo: missing e in Announcement

File: src/bot/AnnouncementListener.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	 * 
+	 * @param client Discord client to operate on
+	 */
+	public async start(client: Discord.Client) {
+		client.on('message', async (message) => {
+			// if no announcement channel set, don't do anything.
+			if (message.channel.id != this.announcementChannel)
+				return;
+
+			let announcementMessage = this.getAnnouncementMessage(message);
```
Comment (tomrijnbeek) [inline src/bot/AnnouncementListener.ts:35]: Are these variables ever mutated? If not, you can use `const` instead of `let`.

**Review by tomrijnbeek — APPROVED** [review]
Code looks fine. Just two style comments. If you can make it pass lint, feel free to squash and merge.

**Comment by ShaneAB** [comment]
Fixed stylistic concerns and passing lint.

**Pull request merged**